packages/opentelemetry-node/lib/processors.js

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright Elasticsearch B.V. and contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// This replicates the SDKs logic of getting processors from env
+// and also adds a new one to collect span metrics like:
+// - metric.otel.sdk.span.live.count
+// - metric.otel.sdk.span.ended.count
+
+const {metrics} = require('@opentelemetry/api');
+const {getStringListFromEnv, getStringFromEnv} = require('@opentelemetry/core');
+const {
+    BatchSpanProcessor,
+    SimpleSpanProcessor,
+    ConsoleSpanExporter,
+} = require('@opentelemetry/sdk-trace-base');
+const {log} = require('./logging');
+
+/**
+ * @typedef {import('@opentelemetry/api').Meter} Meter
+ * @typedef {import('@opentelemetry/api').UpDownCounter} UpDownCounter
+ * @typedef {import('@opentelemetry/api').Counter} Counter
+ * @typedef {import('@opentelemetry/sdk-trace-base').SpanProcessor} SpanProcessor
+ * @typedef {import('@opentelemetry/sdk-trace-base').SpanExporter} SpanExporter
+ */
+
+// @ts-ignore - compiler options do not allow lookp outside `lib` folder
+const ELASTIC_PKG = require('../package.json');
+const ELASTIC_SDK_VERSION = ELASTIC_PKG.version;
+const ELASTIC_SDK_SCOPE = ELASTIC_PKG.name;
+
+const otlpPkgPreffix = '@opentelemetry/exporter-trace-otlp-';
+const otlpProtocol =
+    getStringFromEnv('OTEL_EXPORTER_OTLP_TRACES_PROTOCOL') ??
+    getStringFromEnv('OTEL_EXPORTER_OTLP_PROTOCOL') ??
+    'http/protobuf';
+
+// NOTE: assuming the meter provider is not going to be replaced once
+// the EDOT is started we can cache the meter and metrics in these vars
+/** @type {Meter} */
+let selfMetricsMeter;
+/** @type {UpDownCounter} */
+let liveSpans;
+/** @type {Counter} */
+let closedSpans;
+
+/**
+ * @returns {Meter}
+ */
+function getSpansMeter() {
+    if (selfMetricsMeter) {
+        return selfMetricsMeter;
+    }
+    selfMetricsMeter = metrics.getMeter(ELASTIC_SDK_SCOPE, ELASTIC_SDK_VERSION);
+    return selfMetricsMeter;
+}
+
+/**
+ * @returns {UpDownCounter}
+ */
+function getLiveSpansCounter() {
+    if (liveSpans) {
+        return liveSpans;
+    }
+    liveSpans = getSpansMeter().createUpDownCounter(
+        'otel.sdk.span.live.count',
```

Comment:
The `METRIC_OTEL_SDK_SPAN_LIVE_COUNT` constant (from `@opentelemetry/semantic-conventions/incubating`) should be copied into "./lib/semconv.js", and then that constant used here.

Oh! Actually that has been deprecated and replaced already with [`otel.sdk.span.live`](https://opentelemetry.io/docs/specs/semconv/otel/sdk-metrics/#metric-otelsdkspanlive). Hence should use `METRIC_OTEL_SDK_SPAN_LIVE`.